src/test/run-make/sanitizer-staticlib-link/Makefile

```diff
@@ -0,0 +1,25 @@
+-include ../tools.mk
+
+# This test builds a staticlib, then an executable that links to it.
+# The staticlib and executable both  are compiled with address sanitizer, 
+# and we assert that a fault in the staticlib is correctly detected.
+
+# NOTE the address sanitizer only supports x86_64 linux and macOS
+
+ifeq ($(TARGET),x86_64-apple-darwin)
+ASAN_SUPPORT=$(SANITIZER_SUPPORT)
+EXTRA_RUSTFLAG=-C rpath
+else
+ifeq ($(TARGET),x86_64-unknown-linux-gnu)
+ASAN_SUPPORT=$(SANITIZER_SUPPORT)
+EXTRA_RUSTFLAG=
+endif
+endif
+
+all:
+ifeq ($(ASAN_SUPPORT),1)
+	$(RUSTC) -g -Z sanitizer=address --crate-type staticlib --target $(TARGET) library.rs
+	$(CC) program.c $(call STATICLIB,library) $(call OUT_EXE,program) -lasan $(EXTRACFLAGS) $(EXTRACXXFLAGS)
```

Comment:
Oh, I see, looks like the staticlib doesn't get `librustc_asan` pulled in at all.  Maybe rustc is missing a place it should be automatically be adding that?

My guess is that your local system has a `libasan` with the same symbols as `librustc_asan`, so the final link is happy, but the CI system `libasan` apparently differs.  (note it only complains about two of those symbols being undefined in the end.)

Can you get the final link to use `-lrustc_asan` instead of `-lasan`?